### PR TITLE
BackTester now allows period to be specified; Passes test cases

### PR DESCRIPTION
**Pull request by tfukaza**
Allow users to specify the date range of backtesting data. 

`BackTester.start()` will take 3 new params:
- `start`: start date and time of backtest, in format "MM-DD-YYYY:HH:MM:SS", or `None` to use the max range. 
- `end`: start date and time of backtest, in format "MM-DD-YYYY:HH:MM:SS", or `None` to use current time
- `period`: The range of backtesting, in format "{N}DAYS". If start is not specified, start will be set to end - period. If end is not specified, end = start + period. 

Some modifications were also made to `_base_storage.py`

As you can see, there is much more work to be done - many edge cases are not handled, and code needs much more organizing. I'm opening a PR early to get some feedback on the changes I made so far as well as suggestions to better organize the code.

### CODE REVIEW

File: harvest/api/_base.py [hunk 1]
```diff
@@ -705,6 +705,9 @@ def occ_to_data(self, symbol: str):
         option_type = "call" if symbol[6] == "C" else "put"
         price = float(symbol[7:]) / 1000
         return sym, date, option_type, price
+
```
Comment (tfukaza) [inline harvest/api/_base.py:708]: The API class will be responsible for keeping track of timestamps.

File: harvest/storage/base_storage.py [hunk 2]
```diff
@@ -166,28 +166,28 @@ def load(
                     return data
             return None
 
-        dt_interval = interval_to_timedelta(interval)
-
-        self.storage_lock.acquire()
-        if interval not in self.storage[symbol]:
```
Comment (tfukaza) [inline harvest/storage/base_storage.py:172]: Disabling this feature, as the `interval_to_timedelta` made it difficult to handle cases where "special" intervals, like `-9`, were needed. These are intervals used by `BackTester` to create data needed for simulations.

File: harvest/trader/tester.py [hunk 3]
```diff
@@ -45,13 +45,17 @@ def __init__(self, streamer=None, config={}):
...
+        self._setup_debugger(debug)
+
     def start(
         self,
         interval: str = "5MIN",
         aggregations: List[Any] = [],
         source: str = "PICKLE",
         path: str = "./data",
-        kill_switch: bool = False,
+        start:str = None,
```
Comment (tfukaza) [inline harvest/trader/tester.py:56]: New parameters

File: tests/test_tester.py [hunk 4]
```diff
@@ -11,41 +11,42 @@
...
-
-    #     self.assertListEqual(minutes, days_agg)
-
-    @classmethod
-    def tearDownClass(self):
-        path = pathlib.Path(self.data_dir)
-        shutil.rmtree(path)
-
+   
+    def tear_up_down(func):
```
Comment (tfukaza) [inline tests/test_tester.py:15]: Revised tests

**Comment by shershah010** [comment]
Just a suggestion, what if you used datetime objects for the start and end times?

**Comment by tfukaza** [comment]
I thought about that too, but my conclusions was that strings are easier for coding beginners than date time objects. I myself keep forgetting all the functions required to instantiate dates and whatnot.

**Review by shershah010 — APPROVED** [review]
Looks good. We should fix all endpoints that take datetime objects and switch them to take strings with some specified format such as [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)

**Pull request merged**